readthedocs/vcs_support/backends/git.py

```diff
@@ -254,8 +256,13 @@ def find_ref(self, ref):
         return ref
 
     def ref_exists(self, ref):
-        code, _, _ = self.run('git', 'show-ref', ref, record_as_success=True)
-        return code == 0
+        try:
+            r = Repo(self.working_dir)
+            if r.commit(ref):
+                return True
+        except BadName:
```

Comment:
Perhaps this should be `except (BadName, ValueError):`. I believe `BadName` is raised if the string is invalid like a string that can't be a commit hash or isn't a branch. `ValueError` is raised if a commit is not found for some reason.